### PR TITLE
Adds support for EVM rpc

### PR DESCRIPTION
**Pull request by Jouzo**
## Summary

- Consolidates inside src/rust directory
- Removes fork of sputnikVM and pulls it as cargo dependency
- Adds gRPC server
- Adds protobuf codegen
- Skeleton for eth RPC implementation

## RPCs

Skeleton for RPC implementation in `rpc/libain_wrapper.cpp`
- eth_call
- eth_accounts

### CODE REVIEW

File: src/defid.cpp [hunk 1]
```diff
@@ -64,6 +66,9 @@ static bool AppInit(int argc, char* argv[])
     bool fRet = false;
 
     util::ThreadRename("init");
+    init_runtime();
```
Comment (Mixa84) [inline src/defid.cpp:69]: Remove initialization of EVM for now. We'll do it in separate branch.

Comment (Jouzo) [reply src/defid.cpp:69]: This is the gRPC server runtime

File: contrib/dockerfiles/x86_64-apple-darwin18.dockerfile [hunk 2]
```diff
@@ -15,9 +15,16 @@ RUN apt install -y software-properties-common build-essential git libtool autoto
 pkg-config bsdmainutils python3 libssl-dev libevent-dev libboost-system-dev \
 libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev \
 libminiupnpc-dev libzmq3-dev libqrencode-dev \
-curl cmake \
+curl cmake unzip \
 python3-dev python3-pip libcap-dev libbz2-dev libz-dev fonts-tuffy librsvg2-bin libtiff-tools imagemagick
 
+# install protobuf
+RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-linux-x86_64.zip
```
Comment (Mixa84) [inline contrib/dockerfiles/x86_64-apple-darwin18.dockerfile:22]:
```suggestion
RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-x86_64.zip
```

**Review by Mixa84 — CHANGES_REQUESTED** [review]

**Review by Mixa84 — APPROVED** [review]

**Comment by prasannavl** [comment]
Merging into feature/evm for now. Would like to work on a few more refinements before merging into master.

**Pull request merged**